### PR TITLE
Redshift support

### PR DESCRIPTION
**Pull request by IstvanM**
Adding redshift support with Postgres SQL alchemy adapter. I tested it with Serverless Redshift on my own AWS account.

The main issue is that ARRAY_AGG is not supported by Redshift. This means we can't group by during discovery and collect all the column values. We need to do `select distinct` and window functions for sampling the data.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ pyarrow = "~8.0.0"
 fastparquet = "~0.8.0"
 sqlparse = "^0.4.2"
 tqdm = "^4.64.1"
-cryptography= ">=38.0.4,<39.1.0"
+cryptography= "==38.0.4"
```
Comment (IstvanM) [inline pyproject.toml:22]: This we should fix later, last week because of this the release was broken for Athena.

File: pyproject.toml [hunk 2]
```diff
@@ -42,10 +42,10 @@ redis =  { version= "^4.4.0", optional=true}
 kaleido = {version = "==0.2.1", optional=true}
 dash-iconify = {version ="^0.1.2", optional=true}
 dash-draggable = {version="^0.1.2", optional=true}
-jsonpickle = {version="^3.0.1", optional=true}
```
Comment (IstvanM) [inline pyproject.toml:45]: Not used anymore

File: mitzu/adapters/redshift_adapter.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import mitzu.adapters.generic_adapter as GA
+from typing import List, Optional
+import mitzu.helper as H
+from sqlalchemy import select, literal, distinct
+
+VALUES_COL_NAME = "values"
+
+
+class RedshiftAdapter(PostgresqlAdapter):
+    def __init__(self, project: M.Project):
```
Comment (medgyes) [inline mitzu/adapters/redshift_adapter.py:18]: wouldn't just inherit the constructor?

File: mitzu/adapters/redshift_adapter.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        super().__init__(project)
+
+    def execute_query(self, query: Any) -> pd.DataFrame:
+        if type(query) != str:
+
+            query = str(query.compile(compile_kwargs={"literal_binds": True}))
+            query = query.replace(
+                "%", "%%"
+            )  # bugfix for pyathena, which has string formatting
```
Comment (medgyes) [inline mitzu/adapters/redshift_adapter.py:27]: if this bug exists in redshift too then please update the comment

Comment (IstvanM) [reply mitzu/adapters/redshift_adapter.py:27]: 👍 thank you ...

**Review by medgyes — APPROVED** [review]

**Pull request merged**